22_pipeslab/pipeslab.txt

```diff
@@ -2,20 +2,44 @@ For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
   Answer: 
+	ls -l|grep $USER| cut -c 4|grep x|wc -w
+	[Q: Based on the commands we know by now is there a shorter way to do 	this? With less pipes?] 
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
   Answer: 
+	ls -l|grep $USER| cut -c 4,7,10|grep x|wc -w
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
+	
+	ls -l | cut -f6-8 -d" "| grep $(date +%b" "%d)| wc -l
```

Comment:
1. cut is not good at handling duplicate delimiters, so if you want to use space as a delimiter you'll need to make sure no duplicate spaces appear on the line. ls -l actually prints a lot of duplicate spaces, so we can use tr to reduce them, i.e.: `ls -l | tr -s ' '`

2. date allows you to control its padding strategy. `date +"%b %-d"` won't pad the day with zeroes

3. You need to use quotes around the $(...)

Combine all and get: `ls -l | tr -s ' ' | cut -d' ' -f6-7|grep "$(date '+%b %d')" | wc -l`